src/metric-engine/src/metadata_region.rs

```diff
@@ -60,23 +77,46 @@ impl MetadataRegion {
 
     /// Add a new column key to metadata.
     ///
-    /// This method won't check if the column already exists.
-    pub fn add_column(
+    /// This method won't check if the column already exists. But
+    /// will return if the column is successfully added.
+    pub async fn add_column(
```

Comment:
```suggestion
    #[must_use]
    pub async fn add_column(
```

Reply:
`Result<T>` is marked as `#[must_use]` by default. Do you want to add this annotation to the internal `bool`? Seems only a global clippy warning can do that, which was previously enabled and removed shortly.